### PR TITLE
Add hyperlink to is-default-class annotation

### PR DESCRIPTION
**Pull request by Kartik494**
Fixes: #34887 

This PR add hyperlink of `storageclass.kubernetes.io/is-default-class` annotation .

### CODE REVIEW

**Review by divya-mohan0209 — COMMENTED** [review]
/approve

**Comment by divya-mohan0209** [comment]
@kubernetes/sig-storage-pr-reviews : Please can you provide a technical signoff on this?

File: content/en/docs/concepts/storage/dynamic-provisioning.md [hunk 1]
```diff
@@ -116,7 +116,7 @@ can enable this behavior by:
   is enabled on the API server.
 
 An administrator can mark a specific `StorageClass` as default by adding the
-`storageclass.kubernetes.io/is-default-class` annotation to it.
+[`storageclass.kubernetes.io/is-default-class`](/docs/reference/labels-annotations-taints/#storageclass-kubernetes-io-is-default-class) annotation to it.
```
Comment (divya-mohan0209) [inline content/en/docs/concepts/storage/dynamic-provisioning.md:119]: just a minor nit, we've typically formatted them like this elsewhere.
```suggestion
`storageclass.kubernetes.io/is-default-class` [annotation](/docs/reference/labels-annotations-taints/#storageclass-kubernetes-io-is-default-class) to it.
```

**Review by divya-mohan0209 — COMMENTED** [review]
This is a minor styling nit.

**Comment by Kartik494** [comment]
@divya-mohan0209  i have addressed the changes . PTAL
Thanks!!

**Comment by jsafrane** [comment]
/lgtm
from sig-storage POV.

**Pull request merged**